packages/office-ui-fabric-react/src/components/Dropdown/Dropdown.base.tsx

```diff
@@ -1027,6 +1027,9 @@ export class DropdownBase extends React.Component<IDropdownInternalProps, IDropd
       if (this.props.onFocus) {
         this.props.onFocus(ev);
       }
+      if (openOnFocus && !hasFocus) {
+        this.setState({ isOpen: true });
```

Comment:
Will do.

Reply:
Just for clarity, multiple `setState` calls shouldn't affect any prior calls to it. They will all be queued for updates after the render pass.

What I suspect is happening is that devs are using `this.state` instead of the `setState` functional argument and inadvertently overwriting previous setState calls with state from `this.state`. Calls to `setState` do not update `this.state` until the next render pass. Using the setState functional arg will ensure the correct version of state is used.

Reply:
I unfortunately don't remember the specific circumstances, but in multiple instances I've spent significant amounts of time trying to debug issues where later calls to `setState` (non-functional version) seemed to overwrite earlier ones from the same method, even if each call updated different properties and didn't rely on values from `this.state`. Next time it happens I'll try and remember to share.